src/renderer/components/commands-action-button.tsx

```diff
@@ -182,14 +186,17 @@ export const GistActionButton = observer(
       const { appState } = this.props;
       appState.activeGistAction = GistActionState.publishing;
 
-      const description = await this.getFiddleDescriptionFromUser();
+      try {
+        const description = await this.getFiddleDescriptionFromUser();
 
-      if (description) {
-        await this.publishGist(description);
-        appState.editorMosaic.isEdited = false;
+        if (description) {
+          if (await this.publishGist(description)) {
+            appState.editorMosaic.isEdited = false;
+          }
+        }
+      } finally {
+        appState.activeGistAction = GistActionState.none;
```

Comment:
Here we need to consider the case where `getFiddleDescriptionFromUser` fails

```suggestion
        if (!description) {
          return;
        }
        
        await this.publishGist(description)
        appState.editorMosaic.isEdited = false;
      } catch (_error) {
        appState.editorMosaic.isEdited = true;
      } finally {
        appState.activeGistAction = GistActionState.none;
```

Reply:
Not sure I understand the benefit of this change. In this PR `getFiddleDescriptionFromUser()` is already wrapped in the try/catch block, and the return value is already checked. I don't see which case is not covered.

I also don't think we should do `appState.editorMosaic.isEdited = true;`, since the files might not be edited, so toggling that to true is incorrect state. For example, saving a fiddle locally would mean `isEdited = false`, but with this suggested code change, if you try to publish after saving it, and the publish fails, then the editor is marked as `isEdited = true`, which isn't correct. The publish step can clear the `isEdited` state, it shouldn't ever toggle it to true.

Reply:
> I also don't think we should do appState.editorMosaic.isEdited = true;, since the files might not be edited, so toggling that to true is incorrect state. For example, saving a fiddle locally would mean isEdited = false, but with this suggested code change, if you try to publish after saving it, and the publish fails, then the editor is marked as isEdited = true, which isn't correct. The publish step can clear the isEdited state, it shouldn't ever toggle it to true.

Thank you for adding the context

> Not sure I understand the benefit of this change. In this PR getFiddleDescriptionFromUser() is already wrapped in the try/catch block, and the return value is already checked. I don't see which case is not covered.

my initial thought was that when `getFiddleDescriptionFromUser` fails, it also requires `appState.editorMosaic.isEdited = false;`